packages/widgets/src/lib/components/InferenceWidget/shared/helpers.ts

```diff
@@ -102,10 +104,10 @@ export async function callInferenceApi<T>(
 	if (waitForModel) {
 		headers.set("X-Wait-For-Model", "true");
 	}
-	if (useCache === false) {
+	if (useCache === false && get(isLoggedIn) === true) {
 		headers.set("X-Use-Cache", "false");
 	}
-	if (isOnLoadCall) {
+	if (isOnLoadCall || get(isLoggedIn) === false) {
 		headers.set("X-Load-Model", "0");
 	}
```

Comment:
You'll need to update the logic inside `ConversationalWidget` as well (it uses `@huggingface.js/inference` to  call the inference API)

Reply:
let's do this change and merge? @mishig25

Reply:
handled in d91a5672e831d62fb9355b32e13b125d662ace3b

Reply:
I think this is where you want to do the change:
https://github.com/huggingface/huggingface.js/blob/3f8a3f3c0b53639e4804bce2ee0871f2a0cd7d6f/packages/widgets/src/lib/components/InferenceWidget/widgets/ConversationalWidget/ConversationalWidget.svelte#L154-L161

Reply:
no?

Reply:
yep thats better 7138b8fc78e80aa2a07ed1f01cdb290ea2804995